### PR TITLE
Add module for trail making test

### PR DESCRIPTION
**Pull request by faraz-hussain**
Added Trail Making Test A and B active task: a predefined task that measures visual attention and task switching.
 
 In a trail making test, the participant is asked to connect a series of circles labelled 1, 2, 3... or
 1, A, 2, B, 3, C... and time to complete the test is recorded.

To be used by BiAffect app currently in development, and may be useful to other studies interested in information about visual search speed, scanning, speed of processing, mental flexibility and executive functioning.

### CODE REVIEW

File: ResearchKit/ActiveTasks/ORKTrailmakingContentView.m [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+#import "ORKHeadlineLabel.h"
+#import "ORKLabel.h"
+
+#import "ORKAccessibility.h"
+#import "ORKHelpers_Internal.h"
+#import "ORKSkin.h"
+
+#import "ORKRoundTappingButton.h"
+
```
Comment (umerkhan-apple) [inline ResearchKit/ActiveTasks/ORKTrailmakingContentView.m]: Add extra new line.

File: ResearchKit/ActiveTasks/ORKTrailmakingContentView.m [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+@interface ORKTrailmakingTestView : UIView {
+    int linesToDraw;
+}
+
+@end
+
```
Comment (umerkhan-apple) [inline ResearchKit/ActiveTasks/ORKTrailmakingContentView.m]: Add extra new line.

File: ResearchKit/ActiveTasks/ORKTrailmakingStepViewController.m [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+#import "ORKActiveStepViewController_Internal.h"
+
+#import "ORKTrailmakingStep.h"
+#import "ORKStep_Private.h"
+#import "ORKResult.h"
+
+#import "ORKHelpers_Internal.h"
+#import "ORKStepViewController_Internal.h"
+
```
Comment (umerkhan-apple) [inline ResearchKit/ActiveTasks/ORKTrailmakingStepViewController.m]: Add extra new line.

File: ResearchKit/ActiveTasks/ORKTrailmakingStepViewController.m [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    
+    self.activeStepView.activeCustomView = _trailmakingContentView;
+    
+    for (ORKRoundTappingButton* b in _trailmakingContentView.tapButtons) {
+        [b addTarget:self action:@selector(buttonPressed:forEvent:) forControlEvents:UIControlEventTouchDown];
+    }
+}
+
+- (void)viewWillLayoutSubviews
+{
```
Comment (umerkhan-apple) [inline ResearchKit/ActiveTasks/ORKTrailmakingStepViewController.m]: Move in-line.

File: ResearchKit/ActiveTasks/ORKTrailmakingStepViewController.m [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+                     @"0.587838,0.630662,0.699324,0.148955,0.479730,0.519164,0.481419,0.155052,0.263514,0.183798,0.076014,0.123693,0.231419,0.722997,0.131757,0.913763,0.878378,0.898955,0.912162,0.533972,0.711149,0.798781,0.854730,0.056620,0.334459,0.056620",
+                     @"0.501689,0.523560,0.256757,0.253054,0.724662,0.185864,0.616554,0.841187,0.535473,0.678883,0.256757,0.424084,0.358108,0.866492,0.111486,0.915358,0.133446,0.583770,0.118243,0.109948,0.478041,0.164049,0.868243,0.079407,0.854730,0.564572",
+                     @"0.587140,0.370435,0.695431,0.851304,0.478849,0.481739,0.483926,0.845217,0.265651,0.817391,0.074450,0.873043,0.233503,0.274783,0.133672,0.087826,0.878173,0.103478,0.920474,0.467826,0.717428,0.202609,0.852792,0.942609,0.314721,0.946087"
+                     ];
+    });
+  return testData;
+}
+
+- (NSArray*)fetchRandomTest
+{
```
Comment (umerkhan-apple) [inline ResearchKit/ActiveTasks/ORKTrailmakingStepViewController.m]: Move in-line.

File: ResearchKit/Common/ORKResult.h [hunk 6]
```diff
@@ -1371,4 +1371,61 @@ ORK_CLASS_AVAILABLE
 
 @end
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.h]: Add extra new line.

File: ResearchKit/Common/ORKResult.h [hunk 7]
```diff
@@ -1371,4 +1371,61 @@ ORK_CLASS_AVAILABLE
...
+
+/**
+ If the button was tapped in error.
+ */
+@property (nonatomic, assign) BOOL incorrect;
+
+@end
+
+/**
+ The `ORKTrailmakingResult` class represents the result of a signature step (`ORKTrailmakingStep`).
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.h]: Add extra new line.

File: ResearchKit/Common/ORKResult.h [hunk 8]
```diff
@@ -1371,4 +1371,61 @@ ORK_CLASS_AVAILABLE
...
+ */
+@property (nonatomic,assign) NSUInteger index;
+
+/**
+ If the button was tapped in error.
+ */
+@property (nonatomic, assign) BOOL incorrect;
+
+@end
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.h]: Add extra new line.

File: ResearchKit/Common/ORKResult.m [hunk 9]
```diff
@@ -2230,3 +2230,107 @@ - (instancetype)copyWithZone:(NSZone *)zone {
 }
 
 @end
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.m]: Add extra new line.

File: ResearchKit/Common/ORKResult.m [hunk 10]
```diff
@@ -2230,3 +2230,107 @@ - (instancetype)copyWithZone:(NSZone *)zone {
...
+- (instancetype)copyWithZone:(NSZone *)zone {
+    ORKTrailmakingResult *result = [super copyWithZone:zone];
+    result.numberOfErrors = self.numberOfErrors;
+    result.taps = ORKArrayCopyObjects(self.taps);
+    return result;
+}
+
+@end
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKResult.m]: Add extra new line.

**Comment by faraz-hussain** [comment]
Requested changes made. Thank you!

**Comment by umerkhan-apple** [comment]
@faraz-hussain Thanks for the updates. It looks like there are still merge conflicts. Could you please resolve those so I can merge the PR? Thanks!

**Comment by faraz-hussain** [comment]
Thank you @umerkhan-apple—outstanding merge conflicts have been resolved, and instructions for the trail making test have been edited slightly for greater clarity.

**Comment by faraz-hussain** [comment]
Lines connecting dots are now redrawn if orientation is changed from
portrait to landscape or vice versa to maintain line coherence

**Comment by faraz-hussain** [comment]
Displays timer and error scores.

**Pull request merged**